diff --git a/src/costmap_cost_tool.cpp b/src/costmap_cost_tool.cpp
index 47539607e..39e7a4a6e 100644
--- a/src/costmap_cost_tool.cpp
+++ b/src/costmap_cost_tool.cpp
@@ -64,6 +64,10 @@ void CostmapCostTool::onInitialize()
     false /* Does not create and spin an internal executor*/);
 }
 
+void CostmapCostTool::updateAutoDeactivate() {
+    // TODO: Add implementation
+}
+
 void CostmapCostTool::activate() {}
 void CostmapCostTool::deactivate() {}
 
diff --git a/src/docking_panel.cpp b/src/docking_panel.cpp
index c8a977730..d109db0a3 100644
--- a/src/docking_panel.cpp
+++ b/src/docking_panel.cpp
@@ -324,6 +324,10 @@ void DockingPanel::save(rviz_common::Config config) const
   Panel::save(config);
 }
 
+void DockingPanel::onStartup() {
+    // TODO: Add implementation
+}
+
 void DockingPanel::onDockingButtonPressed()
 {
   auto is_action_server_ready =
