diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4ae1a98ad..3a574800b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,10 @@ find_package(tf2 REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(visualization_msgs REQUIRED)
 
+find_package(ode REQUIRED)
+list(REMOVE_ITEM OMPL_LIBRARIES "ode_double")
+list(APPEND OMPL_LIBRARIES "ODE::ODE")
+
 nav2_package()
 
 if(MSVC)
@@ -68,6 +72,7 @@ target_link_libraries(${library_name}_common PRIVATE
 # Hybrid plugin
 add_library(${library_name} SHARED
   src/smac_planner_hybrid.cpp
+  src/node_hybrid.cpp
 )
 target_include_directories(${library_name}
   PUBLIC
@@ -126,6 +131,7 @@ target_link_libraries(${library_name}_2d PRIVATE
 # Lattice plugin
 add_library(${library_name}_lattice SHARED
   src/smac_planner_lattice.cpp
+  src/node_lattice.cpp
 )
 target_include_directories(${library_name}_lattice
   PUBLIC
diff --git a/include/nav2_smac_planner/a_star.hpp b/include/nav2_smac_planner/a_star.hpp
index 8b127960c..a5dbbc262 100644
--- a/include/nav2_smac_planner/a_star.hpp
+++ b/include/nav2_smac_planner/a_star.hpp
@@ -26,7 +26,6 @@
 #include "nav2_costmap_2d/costmap_2d.hpp"
 #include "nav2_core/planner_exceptions.hpp"
 
-#include "nav2_smac_planner/thirdparty/robin_hood.h"
 #include "nav2_smac_planner/analytic_expansion.hpp"
 #include "nav2_smac_planner/node_2d.hpp"
 #include "nav2_smac_planner/node_hybrid.hpp"
@@ -47,7 +46,7 @@ class AStarAlgorithm
 {
 public:
   typedef NodeT * NodePtr;
-  typedef robin_hood::unordered_node_map<uint64_t, NodeT> Graph;
+  typedef std::unordered_map<uint64_t, NodeT> Graph;
   typedef std::vector<NodePtr> NodeVector;
   typedef std::pair<float, NodeBasic<NodeT>> NodeElement;
   typedef typename NodeT::Coordinates Coordinates;
