diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1f6bd0d9..2f5b15b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,10 +2,9 @@ cmake_minimum_required(VERSION 3.5)
 project(ur_calibration)
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra -Wpedantic)
+  add_compile_options(-Wall -Wextra -Wpedantic -Wno-unused-parameter)
 endif()
-
-add_compile_options(-Wno-unused-parameter)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
   message("${PROJECT_NAME}: You did not request a specific build type: selecting 'RelWithDebInfo'.")
@@ -37,7 +36,7 @@ target_include_directories(calibration
 target_link_libraries(calibration
   ur_client_library::urcl
   Eigen3::Eigen
-  yaml-cpp
+  yaml-cpp::yaml-cpp
   rclcpp::rclcpp
   ur_robot_driver::ur_robot_driver_log_handler
 )
